### PR TITLE
✨(frontend) send username to lti consumer if he/she login

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose
We would like to monitor which watch marsha video on Richie. In order to achieve this, we pass the username of the logged in user through the LTI context.

## Proposal

- [x] Pass username to the lti consumer plugin if a user is logged in on Richie

### CODE REVIEW

File: src/frontend/js/components/LtiConsumer/index.tsx [hunk 1]
```diff
@@ -19,7 +21,12 @@ const LtiConsumer = ({ id }: LtiConsumerProps) => {
   useAsyncEffect(async () => {
     await fetch(`/api/v1.0/plugins/lti-consumer/${id}/context/`)
       .then(checkResponseStatus)
-      .then(setContext)
+      .then((ltiContext: LtiConsumerContext) =>
+        setContext({
+          ...ltiContext,
+          ...(user && { lis_person_sourcedid: user.username }),
```
Comment (sampaccoud) [inline src/frontend/js/components/LtiConsumer/index.tsx:27]: maybe add a test if easy?

Comment (jbpenrath) [reply src/frontend/js/components/LtiConsumer/index.tsx:27]: Yes of course, I just quickly put the code on github.

File: src/frontend/js/components/LtiConsumer/index.tsx [hunk 2]
```diff
@@ -19,7 +21,15 @@ const LtiConsumer = ({ id }: LtiConsumerProps) => {
   useAsyncEffect(async () => {
     await fetch(`/api/v1.0/plugins/lti-consumer/${id}/context/`)
       .then(checkResponseStatus)
-      .then(setContext)
+      .then((ltiContext: LtiConsumerContext) =>
+        setContext({
+          ...ltiContext,
+          content_parameters: {
+            ...ltiContext.content_parameters,
+            ...(user && { lis_person_sourcedid: user.username }),
```
Comment (lunika) [inline src/frontend/js/components/LtiConsumer/index.tsx:29]: should we also send the email or not ?

File: src/frontend/js/components/LtiConsumer/index.spec.tsx [hunk 3]
```diff
@@ -180,4 +199,76 @@ describe('components/LtiConsumer', () => {
...
+      expect(input).toBeInstanceOf(HTMLInputElement);
+      if (input) {
+        expect(input.getAttribute('value')).toEqual(value);
+      }
+    });
+
+    // Check that `lis_person_sourcedid` input exists when user is logged in
+    const input = container.querySelector(`input[name=lis_person_sourcedid]`);
+    expect(input).toBeInstanceOf(HTMLInputElement);
+    if (input) {
```
Comment (lunika) [inline src/frontend/js/components/LtiConsumer/index.spec.tsx:270]: This if will never happen if previous `expect` fails. I suggest to remove this `if` and act like if `input` is not `null`.

File: src/frontend/js/components/LtiConsumer/index.spec.tsx [hunk 4]
```diff
@@ -180,4 +199,76 @@ describe('components/LtiConsumer', () => {
...
+      expect(input.getAttribute('value')).toEqual(username);
```
Comment (lunika) [inline src/frontend/js/components/LtiConsumer/index.spec.tsx:271]:
```suggestion
      expect(input!.getAttribute('value')).toEqual(username);
```

File: src/frontend/js/components/LtiConsumer/index.tsx (hunk 2 above)
Comment (jbpenrath) [reply src/frontend/js/components/LtiConsumer/index.tsx:29]: I don't have this information right now. Without using a custom route to retrieve user information as we will do with fonzie, currently the only one solution is to make two request to OpenEdX:
- first request `/me` to retrieve the username of the logged in user
- then a second request to retrieve account information from this `username`

Comment (lunika) [reply src/frontend/js/components/LtiConsumer/index.tsx:29]: Ok the username is enough for now.

**Review by kernicPanel — APPROVED** [review]

**Review by lunika — APPROVED** [review]

**Pull request merged**